### PR TITLE
fix: chain agnostic on profile

### LINKED ISSUE #3126: Reworking profile to be chain agnostic
- I guess identity is fetched from Kusama? Should we fetch it from Kusama till now ? as BSX doesn't have identity

image

- Basilisk seems doesn't have subscan instance so we can't redirect them to URL so we can hide block explorer links once they will have block explorer
- Also under name iirc

image

### Ref
- #2071
- #226 
- #1322

### PR DESCRIPTION
**Pull request by KngZhi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #3126
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer?target=DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

#### BSX

image

#### RMRK

image

### CODE REVIEW

**Comment by KngZhi** [comment]
@yangwao maybe hide top *Viki* and left side *Viki* on sbx chain?

**Comment by yangwao** [comment]
> top _Viki_ and left side _Viki_ on sbx chain?

most probably yes for now :)

**Comment by KngZhi** [comment]
> > top _Viki_ and left side _Viki_ on sbx chain?
> 
> most probably yes for now :)

hmm, so hide all those identities?

**Comment by yangwao** [comment]
which you mean?

**Review by vikiival — CHANGES_REQUESTED** [review]
.

**Comment by KngZhi** [comment]
> force to show address instead of identity name when user is on the other chain than KSM

For the top, I think your suggestion is good, to conform UIly consistency. For the left side, I'd suggest hiding that, since there is no need to show addr twice.

> for the block explorer it would be nice to have a sort of config that says if the blockexplorer is visible based on the prefix (imagine there will be 2 more chains)

Initially, this is just a quick fix to avoid bsx error, but yeah I can add that.

File: components/rmrk/Profile/ProfileDetail.vue [hunk 1]
```diff
@@ -533,8 +544,16 @@ export default class Profile extends mixins(
     })
   }
 
-  get isBsx(): boolean {
-    return this.urlPrefix === 'bsx'
+  get isAllowSetIdentity(): boolean {
```
Comment (vikiival) [inline components/rmrk/Profile/ProfileDetail.vue:547]: Move to utils pls

Comment (KngZhi) [reply components/rmrk/Profile/ProfileDetail.vue:547]: this is a `getter` only ProfileDetail use.

File: components/rmrk/Profile/ProfileDetail.vue [hunk 2]
```diff
@@ -532,6 +544,18 @@ export default class Profile extends mixins(
...
+  }
+
+  get hasBlockExplorer(): boolean {
+    return Object.keys(this.supportedExplorers).includes(this.urlPrefix)
+  }
+
+  get explorer() {
+    return this.supportedExplorers[this.urlPrefix]
+  }
+
```
Comment (vikiival) [inline components/rmrk/Profile/ProfileDetail.vue:558]: move to utils

Comment (KngZhi) [reply components/rmrk/Profile/ProfileDetail.vue:558]: e, Viki, I am a little confused here. Those are getter that relies on `this.urlPrefix` injected by mixin, so, you mean, move to `PrefixMixin`?

Comment (vikiival) [reply components/rmrk/Profile/ProfileDetail.vue:558]:
```ts
const supportedExplorers: Record<string, string> = {
  rmrk: `[kusama](https://kusama.subscan.io`,
  statemine: `https://statemine.statescan.io`
 }

const hasExplorer = (prefix: string): boolean => {
 return supportedExplorers[prefix] !== undefined
}

const getExlorer = (prefix: string) => {
  return supportedExplorers[prefix]
}
```

then use it in vue.

**Review by vikiival — CHANGES_REQUESTED** [review]

**Comment by vikiival** [comment]
I just want to avoid to have logic in components

**Review by vikiival — APPROVED** [review]
LGTM

**Comment by vikiival** [comment]
pay 60 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $60 @ 52.44 USD/KSM ~ 1.144 $KSM
🧗 [DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc](https://kusama.subscan.io/account/DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc)
🔗 [0x1704b6f26a6a74a1a391d803d262b9ce3b0ad813f9ed9b6e805d49e73d6c71a3](https://kusama.subscan.io/extrinsic/0x1704b6f26a6a74a1a391d803d262b9ce3b0ad813f9ed9b6e805d49e73d6c71a3)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Review by prachi00 — APPROVED** [review]

**Pull request merged**